torchtext/data/functional.py

```diff
@@ -151,3 +153,53 @@ def simple_space_split(iterator):
 
     for line in iterator:
         yield line.split()
+
+
+def read_text_iterator(path, tokenizer):
+    r"""Read text from path and yield a list of tokens based on the tokenizer
+
+    Arguments:
+        path: the file path.
+        tokenizer: the tokenizer used to tokenize string text.
+
+    Examples:
+        >>> from torchtext.data.functional import read_text_iterator
+        >>> tokenizer = get_tokenizer("basic_english")
+        >>> list((read_text_iterator('.data/ptb.train.txt', tokenizer)))
+            [['Sentencepiece', 'encode', 'as', 'pieces'], ['example', 'to', 'try!']]
+    """
+
+    with io.open(path, encoding="utf8") as f:
+        reader = unicode_csv_reader(f)
+        for row in reader:
+            tokens = tokenizer(' '.join(row))
+            yield tokens
+
+
+def create_data_from_iterator(vocab, iterator, removed_tokens=None):
+    r"""Yield a list of ids from an token iterator with a vocab.
+
+    Arguments:
+        vocab: the vocabulary convert token into id.
+        iterator: the iterator yield a list of tokens.
+        removed_tokens: removed tokens from output dataset (Default: None)
+
+    Examples:
+        >>> from torchtext.data.functional import simple_space_split
+        >>> from torchtext.data.functional import create_data_from_iterator
+        >>> vocab = {'Sentencepiece' : 0, 'encode' : 1, 'as' : 2, 'pieces' : 3}
+        >>> ids_iter = create_data_from_iterator(vocab,
+        >>>                               simple_space_split(["Sentencepiece as pieces",
+        >>>                                                   "as pieces"]))
+        >>> for ids in ids_iter:
+        >>>     print([num for num in ids])
+        >>> [0, 2, 3]
+        >>> [2, 3]
+    """
+
+    for tokens in iterator:
+        if removed_tokens is None:
+            yield iter(vocab[token] for token in tokens)
+        else:
+            tokens = list(filter(lambda x: x not in removed_tokens, tokens))
```

Comment:
I think you can do this on the fly :). filter([...]) should be able to feed into iter and simply skip undesired items.

Reply:
Fixed. Thanks.